pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -79,272 +107,246 @@ func newPriorityAssigner(initialOFPriorityFunc InitialOFPriorityGetter, isSingle
-	of, priority := pa.getNextOccupiedOFPriority(ofPriority)
-	return of, of == nil || p.Less(*priority)
-}
-
-// checkLowerBound returns if the Priorities before the input ofPriority are lower than the input Priority,
-// as well as the last occupied ofPriority.
-func (pa *priorityAssigner) checkLowerBound(ofPriority uint16, p types.Priority) (*uint16, bool) {
-	of, priority := pa.getLastOccupiedOFPriority(ofPriority)
-	return of, of == nil || priority.Less(p)
+// GetOFPriority returns if the Priority is registered with the priorityAssigner,
+// and retrieves the corresponding ofPriority.
+func (pa *priorityAssigner) GetOFPriority(p types.Priority) (uint16, bool) {
+	of, registered := pa.priorityMap[p]
+	return of, registered
 }
 
-// getInsertionPoint searches for the ofPriority to insert the input Priority in the table.
-// It is guaranteed that the Priorities before the insertionPoint index is lower than the input Priority,
-// and Priorities *on* and after the insertionPoint index is higher than the input Priority.
-// ofPriority returned will range from PolicyBottomPriority to PolicyTopPriority+1.
-func (pa *priorityAssigner) getInsertionPoint(p types.Priority) (uint16, bool) {
-	insertionPoint := pa.initialOFPriorityFunc(p, pa.isSingleTier)
-	occupied, upwardSearching := false, false
-Loop:
-	for insertionPoint >= PolicyBottomPriority && insertionPoint <= PolicyTopPriority {
-		upperBound, upperOk := pa.checkUpperBound(insertionPoint, p)
-		lowerBound, lowerOk := pa.checkLowerBound(insertionPoint, p)
-		switch {
-		case upperOk && lowerOk:
-			if _, occupied = pa.ofPriorityMap[insertionPoint]; occupied && !upwardSearching {
-				// It might be the case that the ofPriority immediately lower than insertionPoint
-				// is unoccupied and fits the insertion criteria (upperOk and lowerOk). So continue
-				// searching instead of returning current insertionPoint, which will cause the
-				// Priority on current insertionPoint to be reassigned.
-				if insertionPoint != PolicyBottomPriority {
-					insertionPoint--
-					continue Loop
-				}
-			}
-			break Loop
-		case upperOk:
-			// the Priority on lowerBound is higher than current Priority.
-			insertionPoint = *lowerBound - 1
-		case lowerOk:
-			// the Priority on upperBound is lower than current Priority.
-			insertionPoint = *upperBound + 1
-			upwardSearching = true
+// RegisterPriorities registers a list of Priorities with the priorityAssigner. It allocates ofPriorities for
+// input Priorities that are not yet registered. It also returns the ofPriority updates if there are reassignments,
+// as well as a revert function that can undo the registration if any error occurred in data plane.
+func (pa *priorityAssigner) RegisterPriorities(priorities []types.Priority) (map[uint16]uint16, func(), error) {
+	// create a zero-length slice with the same underlying array
+	prioritiesToRegister := priorities[:0]
```

Comment:
just want to make sure that you do intend to overwrite the input `priorities` slice here? you may want to mention this in the function comment since it can be a bit surprising when calling the function

Reply:
Yes. Updated the comment to reflect that.